full-service/src/service/wallet_impl.rs

```diff
@@ -595,6 +595,30 @@ impl<
         ))
     }
 
+    pub fn get_proofs(
+        &self,
+        transaction_log_id: &str,
+    ) -> Result<Vec<JsonProof>, WalletServiceError> {
+        let transaction_log = self.get_transaction(&transaction_log_id)?;
+        let proofs: Vec<JsonProof> = transaction_log
+            .output_txo_ids
+            .iter()
+            .map(|txo_id| {
+                let proof = self
+                    .get_txo(txo_id)
+                    .expect("Could not get Txo")
```

Comment:
Does that make sense? If the user sends an invalid id it will crash the service, no?

Reply:
Whoops - I hadn't pushed my latest commit which has the error handling. Updated now!